tests/integration/Core/Framework/Seo/SeoUrlPersisterTest.php

```diff
@@ -425,6 +425,72 @@ public function testUpdaterDoesNotTouchOtherUrlsFromOtherSalesChannels(): void
         static::assertFalse($seoUrl->getIsDeleted());
     }
 
+    public function testUpdateSeoUrlForDifferentSalesChannelsWithSameSeoPathInfo(): void
+    {
+        $context = Context::createDefaultContext();
+
+        $salesChannelAId = Uuid::randomHex();
+        $this->createStorefrontSalesChannelContext($salesChannelAId, 'test a');
+
+        $salesChannelBId = Uuid::randomHex();
+        $this->createStorefrontSalesChannelContext($salesChannelBId, 'test b');
+
+        $fk = Uuid::randomHex();
+
+        $seoUrlUpdates = [
+            [
+                'foreignKey' => $fk,
+                'salesChannelId' => $salesChannelAId,
+                'pathInfo' => 'normal/path',
+                'seoPathInfo' => 'fancy-path',
+                'isCanonical' => true,
+            ],
+        ];
+        $fks = array_column($seoUrlUpdates, 'foreignKey');
+        $this->seoUrlPersister->updateSeoUrls($context, 'r', $fks, $seoUrlUpdates, $this->salesChannel);
+
+        $seoUrlUpdates = [
+            [
+                'foreignKey' => $fk,
+                'salesChannelId' => $salesChannelBId,
+                'pathInfo' => 'normal/path',
+                'seoPathInfo' => 'fancy-path',
+                'isCanonical' => true,
+            ],
+        ];
+        $fks = array_column($seoUrlUpdates, 'foreignKey');
+        $this->seoUrlPersister->updateSeoUrls($context, 'r', $fks, $seoUrlUpdates, $this->salesChannel);
+
+        $seoUrlUpdates = [
+            [
+                'foreignKey' => $fk,
+                'salesChannelId' => $salesChannelAId,
+                'pathInfo' => 'normal/path',
+                'seoPathInfo' => 'fancy-path-2',
+                'isCanonical' => true,
+            ],
+        ];
+        $fks = array_column($seoUrlUpdates, 'foreignKey');
+        $this->seoUrlPersister->updateSeoUrls($context, 'r', $fks, $seoUrlUpdates, $this->salesChannel);
+
+        $seoUrlUpdates = [
+            [
+                'foreignKey' => $fk,
+                'salesChannelId' => $salesChannelBId,
+                'pathInfo' => 'normal/path',
+                'seoPathInfo' => 'fancy-path-2',
+                'isCanonical' => true,
+            ],
+        ];
+        $fks = array_column($seoUrlUpdates, 'foreignKey');
+        $this->seoUrlPersister->updateSeoUrls($context, 'r', $fks, $seoUrlUpdates, $this->salesChannel);
+
+        $criteria = (new Criteria())->addFilter(new EqualsFilter('routeName', 'r'));
+        /** @var SeoUrlCollection $result */
+        $result = $this->seoUrlRepository->search($criteria, $context)->getEntities();
+        static::assertCount(4, $result);
```

Comment:
we should also check, that the URLs have the right path for the specific saleschannels. and why are there now four? shouldn't there only two?